pyccel/codegen/printing/fcode.py

```diff
@@ -206,6 +212,37 @@ def __init__(self, parser, prefix_module = None):
 
         self.prefix_module = prefix_module
 
+    def check_restrictions(self):
+        """Checks if the namespace contains any of the shortcuts"""
+        valid = [True] * len(iso_c_binding_shortcuts)
+        for variableName in self._namespace.variables:
+            for idx, alias in enumerate(iso_c_binding_shortcuts):
+                if variableName == alias:
+                    valid[idx] = False
+        for functionName in self._namespace.functions:
+            for idx, alias in enumerate(iso_c_binding_shortcuts):
+                if functionName == alias:
+                    valid[idx] = False
+        for variableName in self._namespace.imports['variables']:
+            for idx, alias in enumerate(iso_c_binding_shortcuts):
+                if variableName == alias:
+                    valid[idx] = False
+        for functionName in self._namespace.imports['functions']:
+            for idx, alias in enumerate(iso_c_binding_shortcuts):
+                if functionName == alias:
+                    valid[idx] = False
+        return valid
+
+    def print_macros(self):
+        """Prints the use line that indicates the macros used"""
+        macro = "use ISO_C_BINDING, only: "
+        rename = []
+        for idx, used in enumerate(self._macros):
+            if used:
+                rename.append(iso_c_binding_shortcuts[idx] + ' => ' + iso_c_bindings[idx])
```

Comment:
If I understood you are only printing the import explicitly if the constant is renamed? Otherwise you are still using `use ISO_C_BINDING`? Does this not still cause collisions? `use MODULE_X` is equivalent to `from MODULE_X import *` in python

Reply:
sorry i did not understand this point.

Reply:
Take the following code:
```
C_INT = 4.6
```
On the master branch this is translated to:
```
program prog_tmp

  use, intrinsic :: ISO_C_BINDING

  implicit none

  real(C_DOUBLE) :: C_INT

  C_INT = 4.6_C_DOUBLE

end program prog_tmp
```
which raises the error:
```
/home/emily/Code/test/tmp/__pyccel__/tmp.f90:7:25:

    3 |   use, intrinsic :: ISO_C_BINDING
      |                   2      
......
    7 |   real(C_DOUBLE) :: C_INT
      |                         1
Error: Symbol ‘c_int’ at (1) conflicts with symbol from module ‘__iso_c_binding’, use-associated at (2)
/home/emily/Code/test/tmp/__pyccel__/tmp.f90:9:2:

    9 |   C_INT = 4.6_C_DOUBLE
      |  1
Error: Named constant ‘c_int’ in variable definition context (assignment) at (1)
```
If I modify the code to import explicitly as follows:
```
program prog_tmp

  use, intrinsic :: ISO_C_BINDING, only: C_DOUBLE

  implicit none

  real(C_DOUBLE) :: C_INT

  C_INT = 4.6_C_DOUBLE

end program prog_tmp
```
then the error disappears because `C_INT` is no longer imported. Your improvements have the potential to make this example work but currently it will raise the same error

Reply:
After investigating this issue, i found out that it all stems from 3 lines of code that print `imports += 'use, intrinsic :: ISO_C_BINDING\n'`in program, module and interface. I simply deleted these and only printed the constants i required and it did not seem to mess anything else since all tests passed.
This fixed the issue you described and outputted the following:
```
program prog_r


  use ISO_C_BINDING, only: f64 => C_DOUBLE
  implicit none

  real(f64) :: C_INT

  C_INT = 4.6_f64

end program prog_r
```
I will include this as well in my next push.

Reply:
Great, just make sure you don't miss this case:
```
C_INT = 4.6
i64 = 3
```

Reply:
works as well!
```
program prog_r


  use ISO_C_BINDING, only: C_INT64_T
  implicit none

  integer(C_INT64_T) :: i64

  i64 = 3_C_INT64_T

end program prog_r
```

Reply:
Perfect :+1: